### PR TITLE
refactor: combine WorkspaceContext and WorkspaceMemberContext to WorkspaceDirectory

### PR DESCRIPTION
**Pull request by dsherret**
1. Shift more methods down to `Workspace`
2. Combine `WorkspaceContext` and `WorkspaceMemberContext` into `WorkspaceDir`

### CODE REVIEW

File: src/deno_json/mod.rs [hunk 1]
```diff
@@ -40,14 +40,6 @@ pub use ts::IgnoredCompilerOptions;
 pub use ts::JsxImportSourceConfig;
 pub use ts::TsConfig;
 
-#[derive(Clone, Debug, Default, Eq, PartialEq)]
-pub enum ConfigFlag {
-  #[default]
-  Discover,
-  Path(String),
-  Disabled,
-}
```
Comment (dsherret) [inline src/deno_json/mod.rs:49]: Moved to the CLI.

File: src/deno_json/mod.rs [hunk 2]
```diff
@@ -173,11 +165,6 @@ impl SerializedLintConfig {
   }
 }
 
-#[derive(Clone, Debug, Default, Hash, PartialEq)]
-pub struct WorkspaceLintConfig {
-  pub report: Option<String>,
-}
```
Comment (dsherret) [inline src/deno_json/mod.rs:179]: Moved to the workspace module.

File: src/deno_json/mod.rs [hunk 3]
```diff
@@ -190,7 +177,7 @@ pub struct LintConfig {
 }
 
 impl LintConfig {
-  pub fn default_with_base(base: PathBuf) -> Self {
+  pub fn new_with_base(base: PathBuf) -> Self {
```
Comment (dsherret) [inline src/deno_json/mod.rs:180]: Renamed to match what's done elsewhere and it's shorter.

File: src/workspace/discovery.rs [hunk 4]
```diff
@@ -123,13 +123,8 @@ impl ConfigFolder {
 
 #[derive(Debug)]
 pub enum ConfigFileDiscovery {
-  None {
-    maybe_vendor_dir: Option<PathBuf>,
-  },
-  Workspace {
-    workspace: WorkspaceRc,
-    maybe_vendor_dir: Option<PathBuf>,
```
Comment (dsherret) [inline src/workspace/discovery.rs:131]: Moved down to `Workspace`.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**